src/core/algorithms/fd/dfd/dfd.cpp

```diff
@@ -20,7 +21,7 @@ void DFD::RegisterOptions() {
 }
 
 void DFD::MakeExecuteOptsAvailable() {
-    MakeOptionsAvailable({config::ThreadNumberOpt.GetName()});
+    MakeOptionsAvailable({config::MaxLhsOpt.GetName(), config::ThreadNumberOpt.GetName()});
```

Comment:
This is a bad way to organize the code, which makes options available. `FDAlgorithm::MakeExecuteOptsAvailable` already makes `max_lhs` available, and this functionality should be reused in all other derived classes instead of overriding `MakeExecuteOptsAvailable` and once again making `max_lhs` available. 
The correct way to do this is to implement `FDAlgorithm::MakeExecuteOptsAvailable` similar to this:
```cpp
class FDAlgorithm {
//...

virtual void MakeExecuteOptsAvailableFDInternal() {};

void MakeExecuteOptsAvailable() final {
     MakeOptionsAvailable({config::MaxLhsOpt.GetName()});
     // This method can be overridden by any derived class if needed.
     // And only specific to the derived class options should be made available inside it
     MakeExecuteOptsAvailableFDInternal();
}
// ...
};
```